examples/vpc-scenario-1/main.tf

```diff
@@ -49,34 +63,30 @@ module "vpc" {
   public_subnet_cidrs = ["${var.public_subnet_cidrs}"]
 }
 
-module "ubuntu-xenial-ami" {
-  source  = "../../modules/ami-ubuntu"
-  release = "14.04"
-}
-
-resource "aws_key_pair" "main" {
-  key_name   = "${var.name}"
-  public_key = "${file(var.ssh_pubkey)}"
+module "vpc-sg" {
+  source      = "../../modules/security-group-base"
+  description = "Test project security group"
+  name        = "${var.name}-vpc-sg"
+  vpc_id      = "${module.vpc.vpc_id}"
 }
 
 # shared security group for SSH
 module "public-ssh-sg" {
   source              = "../../modules/ssh-sg"
-  name                = "${var.name}"
-  vpc_id              = "${module.vpc.vpc_id}"
-  allowed_cidr_blocks = "0.0.0.0/0"
+
+  security_group_id = "${module.vpc-sg.id}"
 }
 
 # shared security group, open egress (outbound from nodes)
 module "open-egress-sg" {
   source = "../../modules/open-egress-sg"
-  name   = "${var.name}"
-  vpc_id = "${module.vpc.vpc_id}"
+
+  security_group_id = "${module.vpc-sg.id}"
 }
 
 resource "aws_instance" "web" {
   ami               = "${module.ubuntu-xenial-ami.id}"
-  count             = "${length(module.vpc.public_subnet_ids)}"
+  count             = "1"
```

Comment:
why this change?

Reply:
@ketzacoatl  It was one way I found to resolve an error that I was getting. 
When count is set to `${length(module.vpc.public_subnet_ids)}` instead of 1 the following error is produced:

```
Error: Error applying plan:

2 error(s) occurred:

* aws_instance.web[1]: 1 error(s) occurred:

* aws_instance.web.1: Error launching source instance: InvalidParameterValue: Value (us-east-2a) for parameter availabilityZone is invalid. Subnet 'subnet-439b4139' is in the availability zone us-east-2b
        status code: 400, request id: 0deb09d8-1f53-40ab-a763-5fbda58905a3

* aws_instance.web[2]: 1 error(s) occurred:

* aws_instance.web.2: Error launching source instance: InvalidParameterValue: Value (us-east-2a) for parameter availabilityZone is invalid. Subnet 'subnet-73b9b13e' is in the availability zone us-east-2c
        status code: 400, request id: 9b6acab4-e5b5-4880-bdbc-3d5f3904a290
```

There is probably a better way to address this error though.